### PR TITLE
updated tests for Home Page

### PR DESCRIPTION
**Pull request by VeronikaDolbina**

### CODE REVIEW

File: ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts [hunk 1]
```diff
@@ -6,18 +6,35 @@ import UrlProvider from '../../../providers/UrlProvider';
...
 import MainSiteLinks from '../../../identifiers/mainSite/MainSiteLinks';
 import Links from '../../../preconditionsData/links/Links';
 import BlogTagPath from '../../../providers/BlogTagPath';
 import {qase} from 'playwright-qase-reporter/dist/playwright';
 import {containerSteps, test, expect} from '../../../fixtures/DesktopMobileSetup';
+import TechnologyStackData from '../../../preconditionsData/technologyStack/TechnologyStackData';
 
 test.beforeEach(async () => {
 	await baseDriverSteps.createsNewBrowserAndGoToUrl(UrlProvider.webSiteUrl());
 });
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts:20]: mb better to separate one file for Content and Actions?

File: ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts [hunk 2]
```diff
@@ -34,77 +51,144 @@ test(
...
+
+		const allSectionTitles = whatWeDoContainer.getByTestId(Container.SectionTitle);
+		const testData = [
+			'Product or solution development',
+			'Dedicated team',
+			'Team augmentation',
+			'Tech audit service',
+		];
+		await expect(allSectionTitles).toHaveText(testData);
+	}
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts:130]: please add CTA text and include in title

File: ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts [hunk 3]
```diff
@@ -34,77 +51,144 @@ test(
...
+		expect(await clutchReviewButton.getAttribute('class')).toContain('active');
+
+		const clutchButtons = await partnerTestimonialsContainer.getByTestId(MainSiteButtons.ClutchReviewArrow).all();
+
+		const buttonMap = new Map([
+			[clutchButtons[0], ClutchReviewLinks.MarkBeare],
+			[clutchButtons[1], ClutchReviewLinks.DerickDaily],
+			[clutchButtons[2], ClutchReviewLinks.DarrenCody],
+			[clutchButtons[3], ClutchReviewLinks.AnonymousMedicalDevice],
+			[clutchButtons[4], ClutchReviewLinks.AnonymousPeerToPeer],
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts:152]: incorrect authors order

File: ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts [hunk 4]
```diff
@@ -170,68 +252,70 @@ test.skip(
 	}
 );
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts:254]: pls add checking of "Request a quote" btn text in "case studies" container

File: ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts [hunk 5]
```diff
@@ -170,68 +252,70 @@ test.skip(
...
+
+test(
 	qase(
 		5069,
 		'Check images in "Working with Businesses Worldwide" container from the "Home" page @desktop @mobile @Regression @HomePage @TSWEB-1006'
 	),
 	async () => {
 		const workingWithBusinessesContainer = driver.getByTestId(HomePage.WorkingWithBusinessesWorldwide);
-		const bigNumberImages = workingWithBusinessesContainer.getByTestId(MainSiteImages.BigNumber);
+		const bigNumberImages = workingWithBusinessesContainer.getByTestId(MainSiteImages.BigNumber).locator('visible=true');
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts:340]: why locator?

File: ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts [hunk 6]
```diff
@@ -247,20 +331,20 @@ test.skip(
...
 		5090,
 		'Check redirect by link in "Working with Businesses Worldwide" container from the "Home" page @desktop @mobile @Regression @HomePage @TSWEB-1006'
 	),
 	async () => {
 		const workingWithBusinessesContainer = driver.getByTestId(HomePage.WorkingWithBusinessesWorldwide);
-		const clutchLink = workingWithBusinessesContainer.getByTestId(MainSiteLinks.Clutch);
-
+		const clutchLink = workingWithBusinessesContainer.getByTestId(MainSiteLinks.Clutch).nth(0);
+	
 		await baseDriverSteps.checkRedirectToPage(clutchLink, Links.ClutchHighlights);
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts:365]: pls replace with checkRedirectToClutch()

File: ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts [hunk 7]
```diff
@@ -247,20 +331,20 @@ test.skip(
...
+test(
 	qase(
 		5090,
 		'Check redirect by link in "Working with Businesses Worldwide" container from the "Home" page @desktop @mobile @Regression @HomePage @TSWEB-1006'
 	),
 	async () => {
 		const workingWithBusinessesContainer = driver.getByTestId(HomePage.WorkingWithBusinessesWorldwide);
-		const clutchLink = workingWithBusinessesContainer.getByTestId(MainSiteLinks.Clutch);
-
+		const clutchLink = workingWithBusinessesContainer.getByTestId(MainSiteLinks.Clutch).nth(0);
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts:363]: pls remove nth(0)

File: ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts [hunk 8]
```diff
@@ -273,7 +357,42 @@ test.skip(
...
+		]);
+
+		const allSectionTitles = howWeBuildCommunicationProcessesContainer.getByTestId(Container.SectionTitle);
+		const testData = [
+			'Agile methodology',
+			'Account management',
+			'B2-C1 English proficiency is held by 60% of engineers',
+		];
+		await expect(allSectionTitles).toHaveText(testData);
+	}
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts:402]: pls also add CTA text (also upd test title)

File: ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts [hunk 9]
```diff
@@ -273,7 +357,42 @@ test.skip(
...
+		'Check award cards in "Techstack Achievements" container from the "Home" page @desktop @mobile @Regression @HomePage @TSWEB-1006'
+	),
+	async () => {
+		const ourApproachContainer = driver.getByTestId(HomePage.TechstackAchievements);
+		const awardCards = ourApproachContainer.getByTestId(Container.AwardCard);
+		await baseDriverSteps.checkImagesVisibility(awardCards, 10);
+	}
+);
+
+test(
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts:417]: pls add test for "Recognition and Media Presence" container

File: ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts [hunk 10]
```diff
@@ -170,68 +252,92 @@ test.skip(
...
+		const briefOverviewOfTechnologiesContainer = driver.getByTestId(HomePage.BriefOverviewOfTechnologies);
+		const navigationTabs = await TechnologyStackData.getTechnologyStackTabsForHomePage(briefOverviewOfTechnologiesContainer);
+		const awardCardCountList = [8,5,5,4,8,5]
+		
+		for (let index = 0; index < navigationTabs.length; index++) {
+			navigationTabs[index].click();
+			const awardCards = briefOverviewOfTechnologiesContainer.getByTestId(Container.AwardCard).locator('visible=true');
+		
+		await baseDriverSteps.checkImagesVisibility(awardCards, awardCardCountList[index]);
+		}
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts:293]: why separated for 2 tests?

**Review by Valerii-Kryvokhyzha — CHANGES_REQUESTED** [review]

File: ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts (hunk 3 above)
Comment (VeronikaDolbina) [reply ui_tests_playwright/automation/test/ui/mainSite/HomePage.test.ts:152]: > incorrect authors order

this test has not been updated, it is skipped for now

File: ui_tests_playwright/automation/test/ui/mainSite/HomePageContent.test.ts [hunk 11]
```diff
@@ -0,0 +1,321 @@
...
+	qase(
+		5622,
+		'Check CTA button in "Case Studies" container from the "Home" page @desktop @mobile @Regression @HomePage @TSWEB-1006'
+	),
+	async () => {
+		const caseStudiesContainer = driver.getByTestId(HomePage.CaseStudies);
+		await expect(caseStudiesContainer.getByTestId(MainSiteButtons.ReadAllCases)).toHaveText('Read all cases');
+	}
+);
+
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/HomePageContent.test.ts:156]: here we can add checking the text of Tecnology Stack tabs

File: ui_tests_playwright/automation/test/ui/mainSite/HomePageActions.test.ts [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+		for (const button of ctaButtons) {
+			await baseDriverSteps.checkScrollToContainerByCtaButtonClick(button, HomePage.GetInTouch);
+		}
+	}
+);
+
+test(
+	qase(
+		5011,
+		'Check the "Enhance Healthcare Strategy: Free Cloud Guide" container from the "Home" page @desktop @mobile @Regression @HomePage @TSWEB-1006'
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/HomePageActions.test.ts:40]: better rename to something with redirect to...

File: ui_tests_playwright/automation/test/ui/mainSite/HomePageActions.test.ts [hunk 13]
```diff
@@ -0,0 +1,202 @@
...
+		const buttonMap = new Map([
+			[clutchButtons[0], ClutchReviewLinks.DerickDaily],
+			[clutchButtons[2], ClutchReviewLinks.MarkBeare],
+		]);
+
+		for (const [button, url] of buttonMap) {
+			await baseDriverSteps.checkRedirectToPage(button, url);
+		}
+	}
+);
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/HomePageActions.test.ts:76]: can we unskip this test and update it?

File: ui_tests_playwright/automation/test/ui/mainSite/HomePageActions.test.ts [hunk 14]
```diff
@@ -0,0 +1,202 @@
...
+			[arrows.nth(9), UrlProvider.urlBuilder(UrlPath.QaAsAServ)],
+			[arrows.nth(10), UrlProvider.urlBuilder(UrlPath.ConsultingServ)],
+		]);
+
+		for (const [arrow, url] of arrowUrlMap) {
+			await baseDriverSteps.checkRedirectToPage(arrow, url, UrlProvider.webSiteUrl());
+		}
+	}
+);
+
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/HomePageActions.test.ts:133]: pls add here test for redirecting to case studies in "Case Studies" container

File: ui_tests_playwright/automation/test/ui/mainSite/HomePageActions.test.ts [hunk 15]
```diff
@@ -0,0 +1,202 @@
...
+		for (let index = 0; index < navigationTabs.length; index++) {
+			navigationTabs[index].click();
+			const awardCards = briefOverviewOfTechnologiesContainer
+				.getByTestId(Container.AwardCard)
+				.locator('visible=true');
+
+			await baseDriverSteps.checkImagesVisibility(awardCards, awardCardCountList[index]);
+		}
+	}
+);
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/HomePageActions.test.ts:155]: mb better to move to content test file?

File: ui_tests_playwright/automation/test/ui/mainSite/HomePageActions.test.ts [hunk 16]
```diff
@@ -0,0 +1,202 @@
...
+		'Check redirect by link in "Working with Businesses Worldwide" container from the "Home" page @desktop @mobile @Regression @HomePage @TSWEB-1006'
+	),
+	async () => {
+		const workingWithBusinessesContainer = driver.getByTestId(HomePage.WorkingWithBusinessesWorldwide);
+		const clutchLink = workingWithBusinessesContainer.getByTestId(MainSiteLinks.Clutch).locator('visible=true');
+
+		await baseDriverSteps.checkRedirectToClutch(clutchLink, Links.ClutchHighlights);
+	}
+);
+
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/HomePageActions.test.ts:186]: pls add here test for redirecting to links in "Recognition and Media Presence" container

File: ui_tests_playwright/automation/test/ui/mainSite/HomePageActions.test.ts [hunk 17]
```diff
@@ -0,0 +1,202 @@
...
+		const companyInsightsContainer = driver.getByTestId(HomePage.CompanyInsights);
+		companyInsightsContainer.getByTestId(MainSiteButtons.SeeAllNews).click();
+
+		await baseDriverSteps.checkUrl(UrlProvider.urlBuilder(BlogTagPath.TechstackNews));
+	}
+);
+
+test.afterEach(async () => {
+	await driver.closeDrivers();
+});
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/HomePageActions.test.ts:202]: pls add test for FAQ expanding and colapsing

**Review by Valerii-Kryvokhyzha — APPROVED** [review]

**Pull request merged**